server/middlewares/userValidations.js

```diff
@@ -16,9 +16,15 @@ class UserValidations {
    */
   static validateSignup(req, res, next) {
     // this function validates the input the user sends when trying to register
-    // this regex removes all the white space characters in the email and username
-    req.body.email = req.body.email.toLowerCase().replace(/\s/g, '');
-    req.body.username = req.body.username.replace(/\s/g, '');
+
+    // check if email and username is defined
```

Comment:
I added this block of code to check if username and email are defined before converting them to lowercase.